### PR TITLE
Implemented test cases for delete tasks from tag

### PR DESCRIPTION
**Pull request by juayhee**
Related to #130

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/DeleteTagFromTaskCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    @Test
+    public void equals() {
+        final AddTagToTaskCommand command = new AddTagToTaskCommand(INDEX_FIRST_PERSON, TAG1);
+
+        // If they are the same objects, they are equal
+        assertTrue(command.equals(command));
+
+    }
```
Comment (AAlghrairy) [inline src/test/java/seedu/address/logic/commands/DeleteTagFromTaskCommandTest.java:91]: Consider adding more assertions here for the `equals()` method to increase test coverage.

Specifically, consider creating an additional assertion to check that two commands of equal `index` and `tagName` return true when using `equals()` on them.

File: src/test/java/seedu/address/logic/parser/DeleteTagFromTaskCommandParserTest.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+                new DeleteTagFromTaskCommand(INDEX_FIRST_PERSON, TAG1);
+        assertParseSuccess(parser, " 1 TAG1", expectedDeleteTagFromTaskCommand);
+    }
+
+    @Test
+    void parse_invalidArgs_throwsParseException() {
+        assertParseFailure(parser,
+                " 1 t/important", // Tags can only be alphanumeric
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteTagFromTaskCommand.MESSAGE_USAGE));
+    }
```
Comment (AAlghrairy) [inline src/test/java/seedu/address/logic/parser/DeleteTagFromTaskCommandParserTest.java:30]: Consider adding another method for testing for out-of-bounds integers.

File: src/test/java/seedu/address/logic/commands/DeleteTagFromTaskCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        showTaskAtIndex(model, TASK_FIRST_INDEX);
+        Index outOfBoundIndex = TASK_SECOND_INDEX;
+        // ensures that outOfBoundIndex is still in bounds of address book list
+        assertTrue(outOfBoundIndex.getZeroBased() < model.getTaskBook().getTaskList().size());
+
+        AddTagToTaskCommand addTagToTaskCommand = new AddTagToTaskCommand(outOfBoundIndex, TAG1);
+
+        assertCommandFailure(addTagToTaskCommand, model, Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+    }
+
```
Comment (AAlghrairy) [inline src/test/java/seedu/address/logic/commands/DeleteTagFromTaskCommandTest.java:83]: Consider adding an additional method for testing for invalid `tagName`.

**Review by AAlghrairy — COMMENTED** [review]
Looks good!

There are a few suggestions for increasing test coverage.

**Comment by juayhee** [comment]
> Looks good!
> 
> There are a few suggestions for increasing test coverage.

Thanks for the suggestions! I have implemented them.

File: src/test/java/seedu/address/logic/commands/DeleteTagFromTaskCommandTest.java (hunk 1 above)
Comment (juayhee) [reply src/test/java/seedu/address/logic/commands/DeleteTagFromTaskCommandTest.java]: Added

File: src/test/java/seedu/address/logic/parser/DeleteTagFromTaskCommandParserTest.java (hunk 2 above)
Comment (juayhee) [reply src/test/java/seedu/address/logic/parser/DeleteTagFromTaskCommandParserTest.java:32]: Added

File: src/test/java/seedu/address/logic/commands/DeleteTagFromTaskCommandTest.java (hunk 3 above)
Comment (juayhee) [reply src/test/java/seedu/address/logic/commands/DeleteTagFromTaskCommandTest.java:92]: Added

**Comment by AAlghrairy** [comment]
Looks Good. Just be sure to fix the checkstyle errors before we merge.

**Pull request merged**